src/mqtt.cpp

```diff
@@ -160,54 +161,64 @@ mqtt_publish(const string& prefix, const OpenZWave::ValueID& v)
         return;
     }
 
-    auto topics = make_value_path(prefix, v);
+    auto topics = make_value_path(opts->mqtt_prefix, v);
 
-    int res = mosquitto_publish(mqtt_client, NULL, topics.first.c_str(),
-            value.size(), value.c_str(), 0, true);
-    if (res != 0) {
-        Log::Write(LogLevel_Error, v.GetNodeId(),
-            "Error while publishing message to MQTT topic '%s'", topics.first.c_str());
-    } else {
-        Log::Write(LogLevel_Debug, v.GetNodeId(), "MQTT PUBLISH: %s -> %s",
-            topics.first.c_str(), value.c_str());
+    if (opts->mqtt_name_topics) {
+        res = mosquitto_publish(mqtt_client, NULL, topics.first.c_str(),
+                value.size(), value.c_str(), 0, true);
+        if (res != 0) {
+            Log::Write(LogLevel_Error, v.GetNodeId(),
+                "Error while publishing message to MQTT topic '%s'", topics.first.c_str());
+        } else {
+            Log::Write(LogLevel_Debug, v.GetNodeId(), "MQTT PUBLISH: %s -> %s",
+                topics.first.c_str(), value.c_str());
+        }
     }
 
-    res = mosquitto_publish(mqtt_client, NULL, topics.second.c_str(),
-            value.size(), value.c_str(), 0, true);
-    if (res != 0) {
-        Log::Write(LogLevel_Error, v.GetNodeId(),
-            "Error while publishing message to MQTT topic '%s'", topics.second.c_str());
-    } else {
-        Log::Write(LogLevel_Debug, v.GetNodeId(), "MQTT PUBLISH: %s -> %s",
-            topics.second.c_str(), value.c_str());
+    if (opts->mqtt_id_topics) {
+        res = mosquitto_publish(mqtt_client, NULL, topics.second.c_str(),
+                value.size(), value.c_str(), 0, true);
+        if (res != 0) {
+            Log::Write(LogLevel_Error, v.GetNodeId(),
+                "Error while publishing message to MQTT topic '%s'", topics.second.c_str());
+        } else {
+            Log::Write(LogLevel_Debug, v.GetNodeId(), "MQTT PUBLISH: %s -> %s",
+                topics.second.c_str(), value.c_str());
+        }
     }
 }
 
 void
-mqtt_subscribe(const string& prefix, const OpenZWave::ValueID& v)
+mqtt_subscribe(const options* opts, const OpenZWave::ValueID& v)
 {
+    int res;
+
     // Ignore read only values
     if (OpenZWave::Manager::Get()->IsValueReadOnly(v)) {
         return;
     }
 
     // Make string representation of changeable parameter
-    auto paths = make_value_path(prefix, v);
+    auto paths = make_value_path(opts->mqtt_prefix, v);
     string ep1 = paths.first + "/set";
     string ep2 = paths.second + "/set";
```

Comment:
Since you've split the logic into 2 parts - it would be better to make minor refactor here - move `ep1` and `ep2` inside appropriate condition like
```cpp
if (opts->mqtt_id_topics) {
    string ep = paths.first + "/set";
    res = mosquitto_subscribe(...);
    ....
}
```
Also consider to move `int res` into each condition - to limit visibility of this variable.